### ISSUE TITLE
请教下训练时grad_norm突然爆炸导致reward下降

### ISSUE THREAD

**Issue by xingzhejun** [XueZeyue/DanceGRPO#55]
你好！非常棒的工作。我在复现的时候（8 H800的setting，在8 H800和8 A800都试验过），会观察到在某些步grad_norm和loss会突然一下‘爆炸’，而reward往往也会在接下来的几步中急速下降，如下图所示，这个问题请问之前遇到过吗？我用的github开源数据集，setting没有改动

<img width="431" height="582" alt="Image" src="https://github.com/user-attachments/assets/a481c853-8f56-4f76-b96e-60849204d800" />

**Comment by XueZeyue** [comment]
不瞒你说，我没用wandb跑，都是只看reward和可视化的。。。- -，但我印象中，reward突然下降只有这几类情况
1. timestep selection调小了
2. batch size不够
3. reward model或者prompt有问题。。

**Comment by XueZeyue** [comment]
你用的啥数据集啊

**Comment by xingzhejun** [comment]
> 你用的啥数据集啊

我用的就是咱们repo里面的HPD数据集

**Comment by XueZeyue** [comment]
那这有点奇怪了。。。我跑的时候咋没这个问题，hunyuan他们跑也没有，https://github.com/Tencent-Hunyuan/MixGRPO/issues/3

**Comment by XueZeyue** [comment]
这个爆炸以后reward会涨回来不

**Comment by xingzhejun** [comment]
> 这个爆炸以后reward会涨回来不，我试试看gradient_accumulation_steps改成24

上面wandb红色那条曲线，是我8 A800，爆炸之后涨回来了；绿色曲线对应H 800, 从reward看爆炸的更彻底一点，然后300step的时候没有涨回来。

**Comment by ICE-Bruce** [comment]
> > 你好，我把max_grad_norm改成了0.01，这是8*H800的结果
> > ![Image](https://github.com/user-attachments/assets/bc89e79a-f513-4515-ab6d-642b0e1ff0eb)
> > 这是settings:
> > ```
> > fastvideo/train_grpo_flux.py \
> > --seed 42 \
> > --pretrained_model_name_or_path data/flux \
> > --vae_model_path data/flux \
> > --cache_dir data/.cache \
> > --data_json_path data/rl_embeddings/videos2caption.json \
> > --gradient_checkpointing \
> > --train_batch_size 2 \
> > --num_latent_t 1 \
> > --sp_size 1 \
> > --train_sp_batch_size 2 \
> > --dataloader_num_workers 4 \
> > --gradient_accumulation_steps 12 \
> > --max_train_steps 302 \
> > --learning_rate 1e-5 \
> > --mixed_precision bf16 \
> > --checkpointing_steps 60 \
> > --allow_tf32 \
> > --cfg 0.0 \
> > --output_dir data/outputs/grpo \
> > --h 720 \
> > --w 720 \
> > --t 1 \
> > --sampling_steps 16 \
> > --eta 0.3 \
> > --lr_warmup_steps 0 \
> > --sampler_seed 1223627 \
> > --max_grad_norm 0.01 \
> > --weight_decay 0.0001 \
> > --use_hpsv2 \
> > --num_generations 12 \
> > --shift 3 \
> > --use_group \
> > --ignore_last \
> > --timestep_fraction 0.6 \
> > --clip_range 1e-4 \
> > --adv_clip_max 5.0 \
> > --use_ema \
> > --ema_decay 0.995 \
> > --init_same_noise
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 这是grad_norm:
> > ![Image](https://github.com/user-attachments/assets/607106ad-fb40-4c13-b393-234991677dce)
> 
> 好的谢谢！看起来是RM不准确导致某些步朝反方向优化步子买的太大，grad_norm就会爆炸，假如不截断就会导致潜在的‘崩溃’。我是在未崩塌的checkpoint上续训，将grad_clip设置为0.005，确实也避免了崩溃的现象。我感觉可能当bsz或卡的数量过小的时候，我们需要grad阶段或者假如kl loss来避免这种现象。话说你有试过加入kl loss的实验吗？

请问grad_norm在0.0003这个数量级正常吗？

**Comment by littletomatodonkey** [comment]
> > 你好，我把max_grad_norm改成了0.01，这是8*H800的结果
> > ![Image](https://github.com/user-attachments/assets/bc89e79a-f513-4515-ab6d-642b0e1ff0eb)
> > 这是settings:
> > ```
> > fastvideo/train_grpo_flux.py \
> > --seed 42 \
> > --pretrained_model_name_or_path data/flux \
> > --vae_model_path data/flux \
> > --cache_dir data/.cache \
> > --data_json_path data/rl_embeddings/videos2caption.json \
> > --gradient_checkpointing \
> > --train_batch_size 2 \
> > --num_latent_t 1 \
> > --sp_size 1 \
> > --train_sp_batch_size 2 \
> > --dataloader_num_workers 4 \
> > --gradient_accumulation_steps 12 \
> > --max_train_steps 302 \
> > --learning_rate 1e-5 \
> > --mixed_precision bf16 \
> > --checkpointing_steps 60 \
> > --allow_tf32 \
> > --cfg 0.0 \
> > --output_dir data/outputs/grpo \
> > --h 720 \
> > --w 720 \
> > --t 1 \
> > --sampling_steps 16 \
> > --eta 0.3 \
> > --lr_warmup_steps 0 \
> > --sampler_seed 1223627 \
> > --max_grad_norm 0.01 \
> > --weight_decay 0.0001 \
> > --use_hpsv2 \
> > --num_generations 12 \
> > --shift 3 \
> > --use_group \
> > --ignore_last \
> > --timestep_fraction 0.6 \
> > --clip_range 1e-4 \
> > --adv_clip_max 5.0 \
> > --use_ema \
> > --ema_decay 0.995 \
> > --init_same_noise
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 这是grad_norm:
> > ![Image](https://github.com/user-attachments/assets/607106ad-fb40-4c13-b393-234991677dce)
> 
> 好的谢谢！看起来是RM不准确导致某些步朝反方向优化步子买的太大，grad_norm就会爆炸，假如不截断就会导致潜在的‘崩溃’。我是在未崩塌的checkpoint上续训，将grad_clip设置为0.005，确实也避免了崩溃的现象。我感觉可能当bsz或卡的数量过小的时候，我们需要grad阶段或者假如kl loss来避免这种现象。话说你有试过加入kl loss的实验吗？

请问这样设置，可视化结果也是正常的吗？
